server/src/main/java/com/google/coffeehouse/util/KnownIdGenerator.java

```diff
@@ -0,0 +1,35 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.util;
+
+/**
+ * A class that generates a known ID.
+ *
+ * <p>Used for constructing an object that alraedy has an ID.
```

Comment:
Hmm, this seems like an odd solution, is there a way to just add a second constructor instead? One that takes a known Id directly?

Reply:
That's a good point! I will implement that instead.